### ISSUE TITLE
Partial parameters should have precedence over parent context

### ISSUE THREAD

**Issue by njust** [sunng87/handlebars-rust#495]
With a template like this 
```handlebars
{{~#*inline "displayName"~}}
    Template:{{name}}
{{/inline}}

{{#each data as |name|}}
    Name:{{name}}
    {{>displayName name="aaaa"}}
{{/each}}
```
And the data 

```json
{
    "data": ["hudel", "test"]
}
```
The expected output is:
```
Name:hudel
Template:aaaa

Name:test
Template:aaaa
```

But with version 4.2.1 the output is:
```
Name:hudel
Template:hudel
Name:test
Template:test
```

**Comment by sunng87** [comment]
@njust could you test if the fix in #496 works for your full scenario?

**Comment by njust** [comment]
Unit test works and the full scenario as well. Thanks for the fix!
